src/umpire/util/AllocationMap.hpp

```diff
@@ -32,87 +32,136 @@
 namespace umpire {
 namespace util {
 
-class AllocationMap;
-class RecordList;
-class RecordListConstIterator;
-
-class AllocationMapConstIterator : public std::iterator<std::forward_iterator_tag, AllocationRecord>
+class RecordList
```

Comment:
This needed to be exposed because AllocationMap uses `MemoryMap<RecordList>`. The iterator also needed to be exposed because it's used inside `AllocationMap::ConstIterator`. If `m_inner_iter` were not exposed by value (e.g. a pointer) then this could be forward declared and put in `AllocationMap.cpp`. That is, however, an uglier solution. Fight the indirection!